components/actions_menu/index.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ComponentProps} from 'react';
+import {connect} from 'react-redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
+
+import {getCurrentUser, getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {getCurrentTeamId, getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+import {appsEnabled, makeAppBindingsSelector} from 'mattermost-redux/selectors/entities/apps';
+
+import {AppBindingLocations} from 'mattermost-redux/constants/apps';
+
+import {isSystemAdmin} from 'utils/utils';
+import {isSystemMessage} from 'mattermost-redux/utils/post_utils';
+import {isCombinedUserActivityPost} from 'mattermost-redux/utils/post_list';
+
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+import {Post} from 'mattermost-redux/types/posts';
+import {DoAppCall, PostEphemeralCallResponseForPost} from 'types/apps';
+import {GlobalState} from 'types/store';
+
+import {openModal} from 'actions/views/modals';
+import {doAppCall, postEphemeralCallResponseForPost} from 'actions/apps';
+
+import ActionsMenu from './actions_menu';
+
+type Props = {
+    post: Post;
+    handleCardClick?: (post: Post) => void;
+    handleDropdownOpened: (open: boolean) => void;
+    isMenuOpen: boolean;
+    location?: ComponentProps<typeof ActionsMenu>['location'];
+};
+
+const getPostMenuBindings = makeAppBindingsSelector(AppBindingLocations.POST_MENU_ITEM);
+
+function mapStateToProps(state: GlobalState, ownProps: Props) {
+    const {post} = ownProps;
+
+    const userId = getCurrentUserId(state);
+    const currentTeam = getCurrentTeam(state) || {};
+
+    const systemMessage = isSystemMessage(post);
+
+    const apps = appsEnabled(state);
+    const showBindings = apps && !systemMessage && !isCombinedUserActivityPost(post.id);
+    const appBindings = showBindings ? getPostMenuBindings(state) : undefined;
```

Comment:
I'm not sure I full understand the issue, but it sounds like there are performance concerns with this feature in search results specifically (or any lists where posts are across multiple channels). 

I have strong feelings that we cannot show the actions menu if there are no actions in it.

For infinite scroll lists, Can we just fetch the bindings for the search results that are initially loaded? And then on each lazy load when the user scrolls, fetch the bindings for the next results? Just an idea. I have no idea if any of that is feasible.

Reply:
That would be better, but still not good. Imagine we load 10 posts. That would be 10 calls to the server. The server then has to get the bindings for those 10 calls, which implies sending a request to all installed apps. If we have 4 installed apps, that is 40 requests. And I am talking with low numbers.

I feel that is we were to go with that approach, it would still be an important performance problem.

Reply:
Is there another option @larkox? I'd still like to avoid showing the menu if it's empty, but I don't have another idea how we might do that while preventing the performance issues you'd identified.

If this is just isolated to search results, can we show the empty menu only for search results? In the channel or thread context, can we still hide the menu if there are no bindings?

Reply:
This will be a problem in any place that you have a list of posts along several channels. That means:
- Search results
- Recent mentions (that are actually a search result)
- Saved Posts

Any other context should be fine to hide the actions button.

I cannot think on any other option to solve this.